### ISSUE TITLE
Creating new Form produces timeout

### ISSUE THREAD

**Issue by rn1988** [barrelstrength/sprout-forms#591]
### Description (screenshots, logs, ...)

Creating a new form takes a long time or even causes the database to stuck. This happens with bigger systems that have a lot of content. The query/process that blocks the system is:
ALTER TABLE `content` ADD `field_defaultField1` text

Sprout Forms adds and removes this field while creating the form. The issue could be related to these changes:
https://github.com/barrelstrength/craft-sprout-forms/issues/493
https://github.com/barrelstrength/craft-sprout-forms/commit/04da308636904a790ffc0028d7e49dd27fb14517

### Steps to reproduce

Go to CP > Sprout Forms > Forms > New Form

### Additional info

- Craft version: 3.6.18
- PHP version: 7.4.16
- Database driver & version:  MySQL 5.7.23
- Plugins & versions: 3.13.6

**Comment by BenParizek** [comment]
@rn1988 Can you confirm which version of Sprout Forms you have installed? The issue you link to above was addressed in release [v3.13.4](https://github.com/barrelstrength/craft-sprout-forms/blob/v3/CHANGELOG.md#3134---2021-03-29). Have you upgraded past that version and run the migrations included in v3.13.4?

The error you link to above was accidentally creating rows in the craft_content table when creating Sprout Forms fields. Do you see any unexpected Sprout Forms field columns in your craft_content table? How many columns does your craft_content table have?

Do you see any errors in Craft's log files that might help us learn more about the issue happening?

**Comment by rn1988** [comment]
Hi @BenParizek 

> Can you confirm which version of Sprout Forms you have installed? 
Can confirm it is 3.13.6 (double-checked it)

> The issue you link to above was addressed in release v3.13.4
Yes, in my investigation I found this change and was thinking that these changes may cause the issue. 

> Have you upgraded past that version and run the migrations included in v3.13.4
Yes,  I can confirm that there are no open migrations (Utilities shows 'No content migrations.')

> Do you see any unexpected Sprout Forms field columns in your craft_content table? 
I only see the 'field_defaultField1' column in the database table. That is the one that gets deleted and created while creating a new form. The command 'ALTER TABLE content ADD field_defaultField1 text' is causing the issue that blocks all other database queries (this is what I can see in the processlist from the MySQL database)

> How many columns does your craft_content table have?
Only 63 columns but ~300k rows

**Comment by BenParizek** [comment]
@rn1988 I've been unable to recreate this yet but I do see us doing something odd in how we modify the Craft content table before we establish a Sprout Forms content table. I'll need to dig into that a bit and can follow up once I know more.

Do you happen to have `allowAdminChanges` set to false? I'm not sure how this would behave but if Craft's field's fall under the changes that Project Config blocks in an environment, perhaps the way we initially create and modify that default field gets caught in some ProjectConfig logic that is being blocked by allowAdminChanges. I'm not convinced that's the issue here but just noting if you are seeing this in one environment and not another.

**Comment by rn1988** [comment]
Hi @BenParizek, thanks for investigating!

> Do you happen to have allowAdminChanges set to false? 
Yes, it is false on staging/production and true in the local environment. I can see in all environments that the alter table of content (adding and deleting) is the bottleneck.

**Comment by jamesmacwhite** [comment]
I am hitting the same issue. Trying to create a new form leads to timeout on our staging and live environments. On my development environment I can sometimes create a new form.

Craft CMS 3.7.46.1
Sprout Forms 3.13.8
MySQL 8

I have 263 columns in the content table 3 of those are random fieldDefault fields.

**Comment by jamesmacwhite** [comment]
I have had some success cloning an existing form which appears to not timeout but does take a while. None of the fields are cloned, but I don't really need that, just to be able to initialise a new form.

**Comment by mmorkt** [comment]
Not sure if it's related but i'm getting a error trying to create a new form:
```
SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The maximum row size for the used table type, not counting BLOBs, is 8126. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs
The SQL being executed was: ALTER TABLE `content` ADD `field_defaultField1_dlseakgz` text
```
Works locally (MySQL 8) but not on my server (MariaDB 10.4)
Craft CMS v3.7.47.1
Sprout Forms v3.13.8

**Comment by BenParizek** [comment]
@mmorkt We fixed a migration in [v3.13.9](https://github.com/barrelstrength/craft-sprout-forms/blob/v3/CHANGELOG.md#3139---2022-11-04) that should address an issue with extraneous fields in the content table. Please update to the latest to trigger that migration and let me know if you continue to see the error.

**Comment by mmorkt** [comment]
thank you @BenParizek, it works!
